### ISSUE TITLE
Allow periodic grabbing of thread name

### ISSUE THREAD

**Issue by davidtgoldblatt** [jemalloc/jemalloc#1797]
When dumping sampled allocations, it would often be handy to include the thread name of the allocator from the perspective of pthread_getname_np. We can add an option to re-set prof.thread.name to the current thread name when we sample. This avoid forcing the user to call a mallctl on threads they don't necessarily control.

**Comment by yinan1048576** [comment]
Just had an offline chat with @binliu19. The default behavior, if this option is turned on, should be to try to fetch the thread name via system call at each sample until we have a valid thread name; afterwards we no longer try to fetch. The "until we have a valid thread name" part means one of the following two cases:
- We successfully fetched the thread name from system call and we successfully allocated a buffer to store it internally; or
- The thread name has successfully been set via an explicit `mallctl` call (which can overwrite a previously existing valid thread name, either fetched via system call or set via a `mallctl` call).

We can optionally keep trying to get the thread name via system call even after we have a valid thread name, but it's not ideal, because:
- thread name rarely changes after being set, and
- it risks overwriting a thread name previously set via an explicit `mallctl` call.

So let's just stop as soon as we've got a valid thread name?
